cryptographic_estimators/MAYOEstimator/mayo_estimator.py

```diff
@@ -0,0 +1,116 @@
+    - ``h`` -- external hybridization parameter (default: 0)
+     - ``excluded_algorithms`` -- a list/tuple of algorithms to be excluded (default: [])
+    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
+    - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
+    - ``bit_complexities`` -- determines if complexity is given in bit operations or basic operations (default 1: in bit)
+
+    """
+
+    def __init__(self, n: int, m: int, o: int, k: int, q: int, memory_bound=inf, **kwargs):
+        super(MAYOEstimator, self).__init__(
+            MAYOAlgorithm, 
+            MAYOProblem(n=n, m=m, o=o, k=k, q=q, memory_bound=memory_bound, **kwargs), 
+            **kwargs
+        )
+        self._estimator_type = "scheme"
+
+    def table(self, show_quantum_complexity=0, show_tilde_o_time=0,
+              show_all_parameters=0, precision=1, truncate=0):
+        """
+        Print table describing the complexity of each algorithm and its optimal parameters
+
+        INPUT:
+
+        - ``show_quantum_complexity`` -- show quantum time complexity (default: false)
+        - ``show_tilde_o_time`` -- show Ō time complexity (default: false)
+        - ``show_all_parameters`` -- show all optimization parameters (default: false)
+        - ``precision`` -- number of decimal digits output (default: 1)
+        - ``truncate`` -- truncate rather than round the output (default: false)
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MAYOEstimator import MAYOEstimator
+            sage: E = MAYOEstimator(n=66, m=64, o=8, k=9, q=16)
+            sage: E.table() # long time
+            +--------------+-------------+----------------+
+            |              |             |    estimate    |
+            +--------------+-------------+-------+--------+
+            | algorithm    | attack_type |  time | memory |
+            +--------------+-------------+-------+--------+
+            | DirectAttack |   forgery   | 144.8 |   99.3 |
+            +--------------+-------------+-------+--------+
+
+            sage: E = MAYOEstimator(n=90, m=56, o=8, k=10, q=16)
+            sage: E.table(show_all_parameters=1)
+            +--------------+-------------+----------------------------+
+            |              |             |          estimate          |
+            +--------------+-------------+------+--------+------------+
+            | algorithm    | attack_type | time | memory | parameters |
+            +--------------+-------------+------+--------+------------+
+            | DirectAttack |   forgery   | 21.4 |   56.9 | {'K': 10}  |
+            +--------------+-------------+------+--------+------------+
+
+            sage: E = MAYOEstimator(n=64, m=60, o=10, k=21, q=16)
```

Comment:
Could you add one test per MAYO parameter set? Meaning that one, test for MAYO1, MAYO2, MAYO3 and  MAYO5?

Maybe we can remove some of the long test below.
Please add the tag  `# long time` for test taking more than  5-10 few seconds.